mslice/tests/pixel_workspace_test.py

```diff
@@ -75,8 +75,8 @@ def test_add_list(self):
         result = self.workspace + list_to_add
         result = result.get_signal()
         self.assertAlmostEqual(0, result[0][0], 8)
-        self.assertAlmostEqual(13, result[1][47], 8)
-        self.assertAlmostEqual(35, result[3][52], 8)
+        self.assertAlmostEqual(59, result[1][47], 8)
+        self.assertAlmostEqual(84, result[3][52], 8)
```

Comment:
I've had to change this unit test, due to the changes in `apply_with_corrected_shape` in `numpy_helper.py`. The `apply_with_corrected_shape` function is intended to allow workspaces to be scaled by `numpy` arrays even if the `numpy` array resulting from a workspace cannot be scaled by the desired `numpy` array as per `numpy`'s broadcasting rules. When adding the list to a workspace, as per this unit test, the `apply_with_corrected_shape` function previously would alter the two intial arrays, despite them both being compatible as per numpy's broadcasting rules. This doesn't seem right to me, so I've changed it. As far as I can tell, this change does not effect any existing operations done in the mslice code base.

Reply:
That sounds reasonable. If numpy's broadcasting rules can be applied to the initial arrays they should not get altered.